backend/server/events/events.go

```diff
@@ -0,0 +1,239 @@
+	Place       string `json:"place"`
+	CoverUrl    string `json:"fb_cover_img"`
+}
+
+// Sets up a ticker to fetch events at an interval.
+func EventFetchTimer() {
+	for range time.Tick(time.Duration(FB_FETCH_INTERVAL * time.Second)) {
+		saveEvents()
+	}
+}
+
+// Fetch events from FB
+func fetchEvents(response *FbResponse) error {
+	// Make a request to FB
+	resp, err := http.Get(
+		fmt.Sprintf(
+			"%s%s?access_token=%s&since=%d",
+			FB_API_PATH,
+			FB_EVENT_PATH,
+			FB_TOKEN,
+			time.Now().Unix()),
+	)
+	if err != nil {
+		return fmt.Errorf("There was an error making a request to FB")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("There was an error making a request to FB")
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("Something went wrong with parsing the FB request")
+	}
+
+	json.Unmarshal(body, &response)
+
+	// Check to make sure no error was captured.
+	if (*response).Error != (FbRespError{}) {
+		return fmt.Errorf("Something went wrong with the FB request")
+	}
+	return nil
+}
+
+// Fetch the cover image for an event from Facebook.
+func fetchCoverImage(id string) (string, error) {
+	resp, err := http.Get(
+		fmt.Sprintf(
+			"%s/%s?fields=cover&access_token=%s",
+			FB_API_PATH,
+			id,
+			FB_TOKEN,
+		),
+	)
+	if err != nil {
+		return "", fmt.Errorf("There was an error making a request to FB")
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("There was an error making a request to FB.")
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("Something went wrong with parsing the FB request.")
+	}
+
+	// Unmarshal the response body to pull the cover image from it
+	var result FbResponse
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return "", fmt.Errorf("There was an issue parsing JSON.")
+	}
+	if result.Error != (FbRespError{}) {
+		return "", fmt.Errorf("Something went wrong with the FB request.")
+	}
+	return result.Cover.CoverUri, nil
+}
+
+// Process the facebook event information before saving it to a new file.
+func saveEvents() {
+	var result FbResponse
+	err := fetchEvents(&result)
+	if err != nil {
+		// do something
+	}
+
+	// Store processed events
+	var processedEvents []Event
+
+	for _, element := range result.Data {
+		if len(element.EventTimes) != 0 {
+			for _, occurrence := range element.EventTimes {
+				cover, err := fetchCoverImage(element.Id)
+				if err != nil {
+					// do something
+				}
+				start, err := iso8601.ParseString(occurrence.Start)
+				if err != nil {
+					// do something
+				}
+				end, err := iso8601.ParseString(occurrence.End)
+				if err != nil {
+					// do something
+				}
+
+				processedEvents = append(processedEvents, Event{
+					Name:        element.Name,
+					Description: element.Description,
+					Start:       start.Unix(),
+					End:         end.Unix(),
+					Id:          element.Id,
+					CoverUrl:    cover,
```

Comment:
Not strictly a security related thing but you should note that Facebook image URLs will expire when they are rotated in the CDN. This might take a long time but eventually the images will break - you might want to store the actual image somewhere we host so it doesn't break.

Reply:
This was a consideration I made, but the image URLs are being pulled every 2 minutes, so it should hopefully be unlikely that it breaks